state/processor.go

```diff
@@ -167,34 +188,42 @@ func (tp *TransactionProcessor) ApplyRewards(layer types.LayerID, miners []types
 			log.Uint64("reward", reward.Uint64()),
 			log.LayerId(uint64(layer)),
 		)
-		tp.globalState.AddBalance(account, reward)
+		tp.AddBalance(account, reward)
 		events.Publish(events.RewardReceived{Coinbase: account.String(), Amount: reward.Uint64()})
 	}
-	newHash, err := tp.globalState.Commit(false)
+	newHash, err := tp.Commit(false)
 
 	if err != nil {
-		tp.Log.Error("db write error %v", err)
+		tp.Log.Error("trie write error %v", err)
 		return
 	}
 
-	tp.addStateToHistory(layer, newHash)
+	err = tp.addStateToHistory(layer, newHash)
+	if err != nil {
+		tp.Log.Error("failed to add state to history: %v", err)
+	}
 }
 
-func (tp *TransactionProcessor) Reset(layer types.LayerID) {
+func (tp *TransactionProcessor) LoadState(layer types.LayerID) error {
 	tp.mu.Lock()
 	defer tp.mu.Unlock()
-	if state, ok := tp.prevStates[layer]; ok {
-		newState, err := New(state, tp.globalState.db)
-
+	if state, err := tp.getLayerStateRoot(layer); err == nil {
+		newState, err := New(state, tp.db)
 		if err != nil {
-			log.Panic("cannot revert- improper state")
+			log.Panic("cannot revert- improper state %v", err)
```

Comment:
```suggestion
			log.Panic("cannot revert- improper state: %v", err)
```
Will make it easier for humans to parse and figure out what to search for.